front/components/members/InvitationModal.tsx

```diff
@@ -0,0 +1,521 @@
+import {
+  Button,
+  ContentMessage,
+  Modal,
+  MovingMailIcon,
+  Page,
+  TextArea,
+  XMarkIcon,
+} from "@dust-tt/sparkle";
+import type {
+  ActiveRoleType,
+  MembershipInvitationType,
+  PlanType,
+  RoleType,
+  UserTypeWithWorkspaces,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { useContext, useState } from "react";
+import { mutate, useSWRConfig } from "swr";
+
+import type { ConfirmDataType } from "@app/components/Confirm";
+import { ConfirmContext } from "@app/components/Confirm";
+import { displayRole, ROLES_DATA } from "@app/components/members/Roles";
+import { RoleDropDown } from "@app/components/members/RolesDropDown";
+import type { NotificationType } from "@app/components/sparkle/Notification";
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import {
+  getPriceWithCurrency,
+  PRO_PLAN_29_COST,
+} from "@app/lib/client/subscription";
+import { MAX_UNCONSUMED_INVITATIONS_PER_WORKSPACE_PER_DAY } from "@app/lib/invitations";
+import { isProPlanCode } from "@app/lib/plans/plan_codes";
+import { isEmailValid } from "@app/lib/utils";
+import type {
+  PostInvitationRequestBody,
+  PostInvitationResponseBody,
+} from "@app/pages/api/w/[wId]/invitations";
+import { handleMembersRoleChange } from "@app/pages/w/[wId]/members";
```

Comment:
This import is not super satisfying. Can we move this in `front/lib/client/members.ts` ?